src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -308,12 +311,13 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 			}
 
 			if (commandType == CMD_UPDATE &&
-				targetEntry->resno == partitionColumn->varattno)
+				TargetEntryChangesValue(targetEntry, partitionColumn, joinTree))
```

Comment:
Sorry, I've been ruined by functional programming.

Totally agree with initializing *joinTree at declaration, but passing it into `TargetEntryChangesValue` makes me, at first glance, unsure of whether and afraid it might be modified by that function.

How about passing `queryTree->jointree` into `TargetEntryChangesValue` or marking that parameter `const` instead?